metadata-ingestion/src/datahub/ingestion/source/bigquery_v2/lineage.py

```diff
@@ -310,9 +263,33 @@ def compute_bigquery_lineage_via_catalog_lineage_api(
             )
             raise e
 
+    def _get_parsed_audit_log_events(self, project_id: str) -> Iterable[QueryEvent]:
+        parse_fn: Callable[[Any], Optional[Union[ReadEvent, QueryEvent]]]
+        if self.config.use_exported_bigquery_audit_metadata:
+            logger.info("Populating lineage info via exported GCP audit logs")
+            bq_client = get_bigquery_client(self.config)
+            entries = self._get_exported_bigquery_audit_metadata(bq_client)
+            parse_fn = self._parse_exported_bigquery_audit_metadata
+        else:
+            logger.info("Populating lineage info via exported GCP audit logs")
+            logging_client = _make_gcp_logging_client(project_id)
+            entries = self._get_bigquery_log_entries(logging_client)
+            parse_fn = self._parse_bigquery_log_entries
+
+        for entry in entries:
+            self.report.num_total_log_entries[project_id] += 1
+            try:
+                event = parse_fn(entry)
+                if event:
+                    self.report.num_parsed_log_entries[project_id] += 1
+                    yield event
+            except Exception as e:
+                logger.warning(f"Unable to parse log entry `{entry}`: {e}")
+                self.report.num_lineage_log_parse_failures[project_id] += 1
+
     def _get_bigquery_log_entries(
         self, client: GCPLoggingClient, limit: Optional[int] = None
-    ) -> Union[Iterable[AuditLogEntry], Iterable[BigQueryAuditMetadata]]:
+    ) -> Union[Iterable[AuditLogEntry]]:
```

Comment:
Think this is a more accurate typing. Technically it doesn't matter because both `AuditLogEntry` and `BigQueryAuditMetadata` are `Any`

Reply:
incredible